### PR TITLE
Update HTTPoison to 1.0

### PR DESCRIPTION
**Pull request by toddholmberg**
HTTPoison 0.13 version requirement conflicts with apps using HTTPoison 1.0.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -26,7 +26,7 @@ defmodule Stripy.Mixfile do
   end
 
   defp deps do
-    [{:httpoison, "~> 0.13"},
+    [{:httpoison, "~> 1.0"},
```
Comment (svileng) [inline mix.exs:29]: Let's make this `~> 0.13 or ~> 1.0` for users still on HTTPoison 0.13.

**Review by svileng — CHANGES_REQUESTED** [review]
Sorry for the delay — please see my comment below.

File: mix.exs (hunk 1 above)
Comment (toddholmberg) [reply mix.exs:29]: @svileng Is it possible to use a conditional in the deps list?

Comment (svileng) [reply mix.exs:29]: @toddholmberg Surprisingly yes. https://github.com/appsignal/appsignal-elixir/blob/develop/mix.exs#L92

**Comment by ddorosz** [comment]
Is there still any plan to implement the conditional dependency?

**Comment by svileng** [comment]
@ddorosz I haven't heard from @toddholmberg so feel free to send a PR (it's a super small change). Alternatively I will try to do it soon-ish.

**Comment by svileng** [comment]
Released as part of `1.2.1` 🎉

**Pull request closed without merging**